### PR TITLE
RM-8985 Replace AppConfigBasedSettingsProvider with IoC implementation

### PR DESCRIPTION
**Pull request by a-ctor**
https://re-motion.atlassian.net/browse/RM-8985

### CODE REVIEW

File: Remotion/Core/Reflection.CodeGeneration.TypePipe/RemotionPipelineRegistry.cs
Comment (martin-herwich) [inline Remotion/Core/Reflection.CodeGeneration.TypePipe/RemotionPipelineRegistry.cs:1]: Clean up using directives.

File: Remotion/Core/Reflection.CodeGeneration.TypePipe/RemotionPipelineRegistry.cs [hunk 1]
```diff
@@ -37,7 +37,8 @@ public class RemotionPipelineRegistry : DefaultPipelineRegistry
     private static IPipeline CreateDefaultPipeline (IEnumerable<IParticipant> defaultPipelineParticipants)
     {
       var remotionPipelineFactory = new RemotionPipelineFactory();
-      var settings = new AppConfigBasedSettingsProvider().GetSettings();
+      // We don't resolve the settings from IoC here as this the "remotion-default-pipeline" and customization is not needed at this point
+      var settings = new PipelineSettingsProvider().GetSettings();
```
Comment (martin-herwich) [inline Remotion/Core/Reflection.CodeGeneration.TypePipe/RemotionPipelineRegistry.cs:41]: I don't understand. The original AppConfigBaseSettingsProvider read its settings from the config file, which could be customized. Why don't we need this any more?

**Review by martin-herwich — CHANGES_REQUESTED** [review]

**Review by martin-herwich — APPROVED** [review]

File: Remotion/Core/Reflection.CodeGeneration.TypePipe/Configuration/AppConfigBasedSettingsProvider.obsolete.cs [hunk 2]
```diff
@@ -41,46 +40,19 @@ namespace Remotion.Reflection.CodeGeneration.TypePipe.Configuration
...
     }
 
-#if !FEATURE_STRONGNAMESIGNING
-    [Obsolete("Strong name signing is not supported and throws PlatformNotSupportedException.", DiagnosticId="SYSLIB0017", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
-#endif
-    public bool ForceStrongNaming
-    {
-      get { return _section.ForceStrongNaming.ElementInformation.IsPresent; }
-    }
+    public bool ForceStrongNaming => throw new NotSupportedException("AppConfigBasedSettingsProvider is no longer supported. (Version 6.0.0)");
```
Comment (MichaelKetting) [inline Remotion/Core/Reflection.CodeGeneration.TypePipe/Configuration/AppConfigBasedSettingsProvider.obsolete.cs:50]: The public members with NotSupportedException also get Obsolete attribute.

File: Remotion/Core/Reflection.CodeGeneration.TypePipe/IPipelineSettingsProvider.cs [hunk 3]
```diff
@@ -1,25 +1,29 @@
...
+using Remotion.TypePipe;
 
-namespace Remotion.Reflection.CodeGeneration.TypePipe.Configuration
+namespace Remotion.Reflection.CodeGeneration.TypePipe
 {
-  public class EnableSerializationWithoutAssemblySavingConfigurationElement : ConfigurationElement
+  /// <summary>
+  /// Defines the API required for retrieving <see cref="PipelineSettings"/>.
+  /// </summary>
+  public interface IPipelineSettingsProvider
```
Comment (MichaelKetting) [inline Remotion/Core/Reflection.CodeGeneration.TypePipe/IPipelineSettingsProvider.cs:25]: I prefer to have a seealso-element in the xmldoc listing the implementations. Helps with doc-navigation.

File: Remotion/Data/DomainObjects.UnitTests/RemotionTestsPipelineRegistry.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+using Remotion.TypePipe.Implementation;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.UnitTests
+{
+  /// <summary>
+  /// Adapts the default <see cref="RemotionPipelineRegistry"/> to enable <see cref="PipelineSettings"/>.<see cref="PipelineSettings.EnableSerializationWithoutAssemblySaving"/>.
+  /// </summary>
+  [ImplementationFor(typeof(IPipelineRegistry), RegistrationType = RegistrationType.Decorator)]
+  public class RemotionTestsPipelineRegistry : DefaultPipelineRegistry
```
Comment (MichaelKetting) [inline Remotion/Data/DomainObjects.UnitTests/RemotionTestsPipelineRegistry.cs:30]: Neceassry. Good choice. Looking forward to deleting them when we drop serialization support.

**Review by MichaelKetting — COMMENTED** [review]
Obso

**Pull request merged**